From 9af4dac19a23ceacc5d4442282a052d139c82604 Mon Sep 17 00:00:00 2001
From: Robert Yang <liezhi.yang@windriver.com>
Date: Thu, 22 May 2014 18:03:00 +0800
Subject: [PATCH] pyanaconda/livecd.py: remove the symlink if it is a
 mountpoint

If the mountpoint is a symlink, we need remove the symlink from the
target image and create a dir for it, otherwise the user customized
partitions would not work.

Upstream-Status: Pending

Signed-off-by: Robert Yang <liezhi.yang@windriver.com>
---
 pyanaconda/livecd.py | 11 +++++++++++
 1 file changed, 11 insertions(+)

diff --git a/pyanaconda/livecd.py b/pyanaconda/livecd.py
index 62999e3..44cc8f0 100644
--- a/pyanaconda/livecd.py
+++ b/pyanaconda/livecd.py
@@ -238,6 +238,17 @@ class LiveCDCopyBackend(backend.AnacondaBackend):
         # unmount the filesystems, except for /
         _setupFilesystems(anaconda.storage.mountpoints, teardown=True)
 
+        # If the mountpoint is a symlink, we need remove the symlink from the
+        # target image and create a dir for it, otherwise the user customized
+        # partitions would not work.
+        for mp in mountpoints:
+            tgt_path = "%s/%s" % (anaconda.rootPath, mp)
+            if os.path.islink(tgt_path):
+                log.info("Removing symlink %s" % tgt_path)
+                os.unlink(tgt_path)
+                log.info("Creating directory %s" % tgt_path)
+                os.mkdir(tgt_path)
+
         # mount all of the filesystems under /mnt so we can copy in content
         _setupFilesystems(anaconda.storage.mountpoints,
                           chroot="/mnt")
-- 
1.8.3.1

